system_tests/superstream_system_test.go

```diff
@@ -0,0 +1,203 @@
+	})
+
+	When("just creating a superstream", func() {
+		BeforeEach(func() {
+			superStreamName = "super-stream-test"
+			vhostName = "super-vhost-1"
+		})
+		It("creates and deletes a superStream successfully", func() {
+			By("creating an exchange")
+			Expect(k8sClient.Create(ctx, superStream, &client.CreateOptions{})).To(Succeed())
+			var exchangeInfo *rabbithole.DetailedExchangeInfo
+			Eventually(func() error {
+				var err error
+				exchangeInfo, err = rabbitClient.GetExchange(vhostName, superStreamName)
+				return err
+			}, 30, 2).Should(BeNil())
+
+			Expect(*exchangeInfo).To(MatchFields(IgnoreExtras, Fields{
+				"Name":       Equal("super-stream-test"),
+				"Vhost":      Equal(vhostName),
+				"Type":       Equal("direct"),
+				"AutoDelete": BeFalse(),
+				"Durable":    BeTrue(),
+			}))
+
+			By("creating n queues")
+			for _, routingKey := range superStream.Spec.RoutingKeys {
+				var qInfo *rabbithole.DetailedQueueInfo
+				Eventually(func() error {
+					var err error
+					qInfo, err = rabbitClient.GetQueue(vhostName, fmt.Sprintf("super-stream-test-%s", routingKey))
+					return err
+				}, 10, 2).Should(BeNil())
+
+				Expect(*qInfo).To(MatchFields(IgnoreExtras, Fields{
+					"Name":       Equal(fmt.Sprintf("super-stream-test-%s", routingKey)),
+					"Vhost":      Equal(vhostName),
+					"AutoDelete": Equal(rabbithole.AutoDelete(false)),
+					"Durable":    BeTrue(),
+					"Type":       Equal("stream"),
+				}))
+			}
+
+			By("creating n bindings")
+			foundPartitionOrderStreams := make(map[int]rabbithole.BindingInfo)
+			for _, routingKey := range superStream.Spec.RoutingKeys {
+				var fetchedBinding rabbithole.BindingInfo
+				Eventually(func() bool {
+					var err error
+					bindings, err := rabbitClient.ListBindingsIn(vhostName)
+					Expect(err).NotTo(HaveOccurred())
+					for _, b := range bindings {
+						if b.Source == "super-stream-test" && b.Destination == fmt.Sprintf("super-stream-test-%s", routingKey) {
+							fetchedBinding = b
+							return true
+						}
+					}
+					return false
+				}, 10, 2).Should(BeTrue(), "cannot find created binding")
+				Expect(fetchedBinding).To(MatchFields(IgnoreExtras, Fields{
+					"Vhost":           Equal(vhostName),
+					"Source":          Equal("super-stream-test"),
+					"Destination":     Equal(fmt.Sprintf("super-stream-test-%s", routingKey)),
+					"DestinationType": Equal("queue"),
+					"RoutingKey":      Equal(routingKey),
+				}))
+				Expect(fetchedBinding.Arguments).To(HaveKey("x-stream-partition-order"))
+				partitionOrder := int(fetchedBinding.Arguments["x-stream-partition-order"].(float64))
+				foundPartitionOrderStreams[partitionOrder] = fetchedBinding
+			}
+			for i := 0; i < superStream.Spec.Partitions; i++ {
+				_, ok := foundPartitionOrderStreams[i]
+				Expect(ok).To(
+					BeTrue(),
+					fmt.Sprintf("Expected to find a partition assigned to all indices, but %d was missing: %+v", i, foundPartitionOrderStreams),
+				)
+			}
```

Comment:
We have the same test at integration level. I'm not sure if this test at system level is adding more coverage. WDYT?

Reply:
Agreed, it made more sense when there was the other CRD but it's less useful now.